visualizer.py

```
@@ -0,0 +1,206 @@
+    draw_info.window.blit(title, (draw_info.width/2 - title.get_width()/2, 5))
+
+    controls = draw_info.FONT.render("R - Reset | SPACE - Start | A - Ascending | D - Descending", 1, draw_info.BLACK)
+    draw_info.window.blit(controls, (draw_info.width/2 - controls.get_width()/2, 35))
+
+    sorting = draw_info.FONT.render("I - Insertion Sort | B - Bubble Sort", 1, draw_info.BLACK)
+    draw_info.window.blit(sorting, (draw_info.width/2 - sorting.get_width()/2, 65))
+
+    draw_list(draw_info)
+    pygame.display.update()
+
+def draw_list(draw_info, color_positions={}, clear_bg = False):
```

Ruff (0.13.3)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)